Libplanet/Blockchain/BlockChain.cs

```diff
@@ -917,13 +917,18 @@ public long GetNextTxNonce(Address address)
                 try
                 {
                     SetStates(block, actionEvaluations, false);
+                    block = new Block<T>(block, trieStateStore.GetRootHash(block.Hash));
+                    SetStates(block, actionEvaluations, false);
```

Comment:
IMO We don't need `SetStates` twice in this statement.

Reply:
It's needed. (it's a quite tricky part.)

In first `SetStates(block, actionEvaluations, false)`, `block.Hash` isn't a real block hash. it can't be determined since `Block<T>.Hash` needs `StateRootHash` from `SetState()`. so we need to do `SetStates` to calculate and confirm `.Hash`, and repeat it to store it.

I think it would be better there is some comment about it. I'll add it.

Reply:
I amended to leave comments. https://github.com/planetarium/libplanet/pull/1116/commits/78d50d6750a3a4f281bb119497bdd42a946ade28

but if you have any questions and/or ideas yet, feel free to left here.

Reply:
I think it is ok to leave `FIXME` comments.